source/Playnite/Scripting/PlayniteScript.cs

```diff
@@ -92,8 +92,10 @@ public static PlayniteScript FromFile(string path)
             {
                 return new IronPythonScript(path);
             }
-            else if (extension == ".ps1")
+            // ps1 and dll PowerShell modules are not supported
+            else if (extension == ".psm1" || extension == ".psd1")

```

Comment:
Can't we completely remove module file requirement from PowerShell extensions? As far as I know you should be able to import module from a directory. At least that's how Import-Module used to work in PowerShell 2. That would also add support for compiled modules.

Reply:
I can look into loading as a directory, that sou ds good. But the problem is extension loading currently detects native dll/IronPython/PowerShell based off the file extension. I can't remove this test here because it's needed to detect if it's PowerShell at all.

Reply:
Also, it looks like using a directory would import _all_ modules in the specified path. The way I wrote this, I need to get a single module as a result and I use it to get it's `ExportedCommands`. I don't think it's too unreasonable for developers to use multiple modules, they may want to bundle an existing PowerShell module as a library they want to use. I don't think the path import works.

Reply:
The directory import should import module based on manifest file, if you have some other modules there, then those should be imported only if the manifest for main module imports them as well. I specifically remember using it that way years ago. But again, that was in PS 2.0 days, maybe they changed it.

I only marked this line because I saw specific mention of psm1 file. Of course the module would have to be kept in our manifest for Python extensions, but it can be removed for PS ones. If it's specified and is .py, then it's Python, otherwise treat it as PS.

Reply:
Okay, it does appear to do the right thing and only import one module, it doesn't import all modules like I was concerned about. However, this _requires_ the name of the module to match the name of the directory. I guess developers can control the name of the extension and this would work, but I think this could be confusing.

```
mkdir moduleTest
echo 'function foo() { echo "foo" }' > moduleTest\foo.psm1
echo 'function bar() { echo "bar" }' > moduleTest\moduleTest.psm1
$module = Import-Module -PassThru .\moduleTest
# Module is from moduleTest.psm1, and foo.psm1 is not imported
```